Fix SQA doco of heat transfer module

GiudGiud opened the pull request:
We saw the failure by adding a new undocumented BC to the heat transfer module.

These changes let the SQA error detection work again for the heat transfer module

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3112031/), step Docs: sync website on f40c601 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/31247/site)

This comment will be updated on new commits.

GiudGiud commented:
ok we have a problem @cticenhour that should have failed SQA?

GiudGiud commented:
on it

GiudGiud commented:
ok that failed as expected. So the failure is specific to heat transfer. 
@NamjaeChoi also said  thermal hydraulics was affected

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/3112971/), step Generate coverage on d19b146 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>17aa62</th>
      <th colspan=4>#31247 d19b14</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.57%</td>
      <td>85.54%</td>
      <td>-0.03%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>116330</td>
      <td>116292</td>
      <td>-38</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>19614</td>
      <td>19652</td>
      <td>+38</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/31247/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/31247/coverage/framework)


## Modules coverage

### Electromagnetics

<table>
  <thead>
    <tr>
      <th></th>
      <th>17aa62</th>
      <th colspan=4>#31247 d19b14</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>96.52%</td>
      <td>92.32%</td>
      <td>-4.20%</td>
      <td>0.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>637</td>
      <td>637</td>
      <td>-</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>23</td>
      <td>53</td>
      <td>+30</td>
      <td>30</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/31247/coverage_diff/electromagnetics)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/31247/coverage/electromagnetics)

### Heat transfer

<table>
  <thead>
    <tr>
      <th></th>
      <th>17aa62</th>
      <th colspan=4>#31247 d19b14</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>89.16%</td>
      <td>88.97%</td>
      <td>-0.19%</td>
      <td>0.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>4608</td>
      <td>4608</td>
      <td>-</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>560</td>
      <td>571</td>
      <td>+11</td>
      <td>11</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/31247/coverage_diff/heat_transfer)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/31247/coverage/heat_transfer)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/31247/coverage/xfem)

</details>

## Warnings

- `electromagnetics` new line coverage rate 0.00% is less than the suggested 90.0%
- `heat_transfer` new line coverage rate 0.00% is less than the suggested 90.0%



This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/3112971/), step Verify coverage on d19b146 wanted to post the following:


The following coverage requirement(s) failed:
- `electromagnetics` coverage rate 92.32% is less than the required 94.0%

GiudGiud committed:
Update SQA doco to heat transfer name

GiudGiud commented:
ok that caught it but also discovered new issues

GiudGiud committed:
Add missing class descriptions
refs #15968

GiudGiud committed:
Remove Modules/ThermalContact syntax
It was only used for thermal contact
and ThermalContact was mostly used without it

NamjaeChoi commented:
> ok that failed as expected. So the failure is specific to heat transfer. @NamjaeChoi also said thermal hydraulics was affected

Thermal hydraulics might be my mistake. I was registering to a wrong app.

GiudGiud committed:
Add parameter groups for GapConductance, when added by the action

GiudGiud committed:
Add missing source documentation to heat transfer module objects

GiudGiud commented:
we need to check the formatting of these pages but otherwise ready for review

GiudGiud commented:
past this round of formatting fix they look ok

cticenhour reviewed (changes requested):
Thanks for triaging and fixing this up. Current CI failures are unrelated. A few suggestions:

GiudGiud commented on modules/heat_transfer/doc/content/source/bcs/CoupledConvectiveFlux.md:
```diff
@@ -1 +1,21 @@
+!syntax description /BCs/CoupledConvectiveFlux
+
+!alert warning
+`CoupledConvectiveFlux` is deprecated.
```
```suggestion
`CoupledConvectiveFlux` is deprecated. Please use [CoupledConvectiveHeatFluxBC.md] instead.
```

GiudGiud committed:
Address Casey's review

- add dot at end of class description
- comma, possibly an Oxford one
- point to replacement object

Co-authored-by: Casey Icenhour <cticenhour@gmail.com>
